torchrl/envs/transforms/transforms.py

```diff
@@ -1885,11 +1894,13 @@ def build_td_for_shared_vecnorm(
             return td_select.memmap_()
         return td_select.share_memory_()
 
-    def get_extra_state(self) -> TensorDictBase:
-        return self._td
+    def get_extra_state(self) -> OrderedDict:
+        return OrderedDict([("lock", self.lock), ("td", self._td)])
```

Comment:
There are a few places in the code where we process state_dicts, for example we call share_memory_ to make shared versions of existing state_dicts. I ran into issues since those pieces of code assumed that state_dicts were recursive OrderedDicts, so inserting a dict rather than an OrderedDict caused failures.

For example EnvCreator.share_memory

It's fixable but will expand the scope of the change, since this OrderedDict was very small this seemed easier but I can go and fix it if you want, please advise.